CONTRIBUTING.md

```diff
@@ -0,0 +1,198 @@
+* Bump to new minor version \(e.g. `1.0.0` to `1.1.0-rc0`\): `bump2version minor`
+* Bump to new major version \(e.g. `1.0.0` to `2.0.0-rc0`\): `bump2version major`
+* Bump to new release candidate version \(e.g. `1.0.0-rc0` to `1.0.0-rc1`\): `bump2version rc`
+* To create a new release version \(e.g. `1.0.0-rc0` to `1.0.0`\): `bump2version release`
+
+### Dependency support
+Packages have to choose which versions of dependencies they officially support,
+with minimum supported versions of each dependency used in continuous
+integration testing. BrainGlobe projects should follow
+[NEP 29 — Recommend Python and NumPy version support as a community policy
+standard](https://numpy.org/neps/nep-0029-deprecation_policy.html) to
+determine the **minimum** set of supported package versions:
+
+- The last 42 months of Python releases
+- The last 24 months of NumPy releases
+
+In addition to this the last 24 months of other dependencies should also be
+supported.
+---
+
+## Cellfinder-specific conventions 
+### Dependencies
+
+Cellfinder includes a lot of functionality brought in from other packages. It is important that 
+each repository focusses on a single task. Changes should be added to the relevant repository:
+
+* **Cell detection in whole-brain microscopy images**: [cellfinder](https://github.com/BrainGlobe/cellfinder)
+* **Cell detection algorithm**: [cellfinder-core](https://github.com/BrainGlobe/cellfinder-core)
+* **Regstration:** [brainreg](https://github.com/brainglobe/brainreg)
+* **Image data IO:** [imio](https://github.com/brainglobe/imio)
+* **General imaging processing \(and other\) tools:**  [imlib](https://github.com/adamltyson/imlib)
+
+### File paths
+
+All file paths should be defined in `cellfinder.tools.prep.Paths`. Any intermediate file paths, \(i.e. those which are not of interest to the typical end user\) should be prefixed with `tmp__`. These should then be cleaned up as soon as possible after generation.
+
+### Releasing `cellfinder-core`
+
+1. Make sure you have [`bump2version`](https://github.com/c4urself/bump2version/#installation) installed.
+2. Check out the latest `main` branch from upstream.
+3. Run `bump2version <type>`, where `<type>` is replaced by one of {major, minor, patch}. This will create a new pre-release commit and tag with "rc" in
+the filename.
+4. Push the result to the cellfinder-core repository using `git push upstream --follow-tags`.
+5. Check that all the tests pass, the package builds correctly, and that all built packages are uploaded to PyPi.
+6. Check that there are no issues with the new version of `cellfinder-core` and `cellfinder` by going to https://github.com/brainglobe/cellfinder/actions/workflows/test_and_deploy.yml, clicking on the "Run workflow" button the left, typing "true" into the prompt, and clicking "Run workflow". This will run the `cellfinder` tests with the development version of `cellfinder-core`. Make sure these tests pass.
```

Comment:
I think hyperlinks without something indicating that they are as such is discouraged
```suggestion
6. Check that there are no issues with the new version of `cellfinder-core` and `cellfinder` by going to <https://github.com/brainglobe/cellfinder/actions/workflows/test_and_deploy.yml/>, clicking on the "Run workflow" button on the left, typing "true" into the prompt, and clicking "Run workflow". This will run the `cellfinder` tests with the development version of `cellfinder-core`. Make sure that these tests pass.
```